apps/dot_ext/loggers.py

```diff
@@ -0,0 +1,211 @@
+import re
+import uuid
+from django.db import transaction
+from django.db.utils import IntegrityError
+from oauth2_provider.models import get_application_model
+from .models import AuthFlowUuid
+
+
+"""
+  Logger related functions for dot_ext/mymedicare_cb modules.
+
+  The AuthFlowUuid tracking object is used to track the
+  auth flow between beneficiary and 3rd party application sessions.
+
+  Values are retrieved/updated in the request.session.
+"""
+
+# List of value keys that are being tracked via request.session
+SESSION_AUTH_FLOW_TRACE_KEYS = ['auth_uuid', 'auth_client_id', 'auth_app_id', 'auth_app_name', 'auth_pkce_method']
+
+# REGEX of paths that should be updated with auth flow info in hhs_oauth_server.request_logging.py
+AUTH_FLOW_REQUEST_LOGGING_PATHS_REGEX = "(^/v1/o/authorize/.*|^/mymedicare/login$|^/mymedicare/sls-callback$|^/v1/o/token/$)"
+
+
+def is_path_part_of_auth_flow_trace(path):
+    '''
+    Check if provided path is related to the authorization flow logging.
+
+    CALLED FROM: hhs_oauth_server.request_logging.RequestResponseLog.__str__()
+    '''
+    return re.match(AUTH_FLOW_REQUEST_LOGGING_PATHS_REGEX, path)
+
+
+def cleanup_session_auth_flow_trace(request):
+    '''
+    Clean up auth flow related items in a session.
+
+    CALLED FROM:  apps.testclient.views.callback()
+                  apps.dot_ext.views.authorization.AuthorizationView.form_valid()
+    '''
+    for k in SESSION_AUTH_FLOW_TRACE_KEYS:
+        try:
+            del request.session[k]
+        except KeyError:
+            pass
+
+
+def create_session_auth_flow_trace(request):
+    '''
+    Create auth flow log tracing related items.
+
+    - Create a new AuthFlowUuid instance.
+    - Set new auth flow values in session.
+
+    CALLED FROM:  apps.dot_ext.views.authorization.AuthorizationView.dispatch()
+    '''
+    # Create new authorization flow trace UUID.
+    new_auth_uuid = str(uuid.uuid4())
+
+    request.session['auth_uuid'] = new_auth_uuid
+
+    client_id_param = request.GET.get("client_id", None)
+    auth_pkce_method = request.GET.get("code_challenge_method", None)
+
+    if client_id_param:
```

Comment:
This is more of an informational question, but is it possible for someone to have a client_id but **not** an application?  Maybe in sandbox but not prod?

Reply:
@njdister All valid `client_id` will have an application the ID belongs to.  I'm thinking if an invalid client ID is specified, that we shouldn't log it in `auth_client_id` and keep the current behavior of setting it to blank. The log entry can be reviewed to see what the attempted `client_id` was in that case.

Reply:
That makes sense, thanks for the clarification!